### PR TITLE
 Blob Maker

### PR DESCRIPTION
**Pull request by AbhineshJha**
# Title and Issue number 

Title : Blob maker
# Checklist:

- [] I have mentioned the issue number in my Pull Request.
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have created a helpful and easy to understand `README.md`
- [] I have updated the Index.html file for my contribution

Screenshot 2023-10-09 140749

### CODE REVIEW

File: native/machine-coding/blobmaker/app.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+function createBlob(){
+    let radiusOne = sliders[0].value;
+    let radiusTwo = sliders[1].value;
+    let radiusThree = sliders[2].value;
+    let radiusFour = sliders[3].value;
+
+    let blobHeight = inputs[0].value;
+    let blobWidth = inputs[1].value;
+
+    let borderRadius = `${radiusOne}% ${100-radiusOne}% ${100 - radiusThree}% ${radiusThree}% / ${radiusFour}% ${radiusTwo}% ${100 - radiusTwo}% ${100 - radiusFour}%`;
```
Comment (arpansaha13) [inline native/machine-coding/blobmaker/app.js:22]: Looks like these variables are never updated. Please use `const` instead of `let`.

Comment (AbhineshJha) [reply native/machine-coding/blobmaker/app.js]: done

Comment (arpansaha13) [inline native/machine-coding/blobmaker/app.js]: Looks like these variables are never updated. Please use `const` instead of `let`.

**Review by arpansaha13 — CHANGES_REQUESTED** [review]

File: native/machine-coding/blobmaker/app.js [hunk 2]
```diff
@@ -11,13 +11,13 @@ let inputs = document.querySelectorAll("input[type='number']");
...
+    const radiusTwo = sliders[1].value;
+    const radiusThree = sliders[2].value;
+    const radiusFour = sliders[3].value;
 
-    let blobHeight = inputs[0].value;
-    let blobWidth = inputs[1].value;
+    const blobHeight = inputs[0].value;
+    const blobWidth = inputs[1].value;
 
     let borderRadius = `${radiusOne}% ${100-radiusOne}% ${100 - radiusThree}% ${radiusThree}% / ${radiusFour}% ${radiusTwo}% ${100 - radiusTwo}% ${100 - radiusFour}%`;
```
Comment (arpansaha13) [inline native/machine-coding/blobmaker/app.js:22]: Please explain the logic behind this line either in comments or a `README.md` file if required.
Also, you can use const here too.

**Review by arpansaha13 — CHANGES_REQUESTED** [review]

File: native/machine-coding/blobmaker/app.js (hunk 2 above)
Comment (AbhineshJha) [reply native/machine-coding/blobmaker/app.js]: added bruoh

Comment (arpansaha13) [reply native/machine-coding/blobmaker/app.js]: Add the explanation of logic too please. The `borderRadius` logic is not readily understandable. So it needs an explanation.

File: native/machine-coding/blobmaker/style.css
Comment (sadanandpai) [inline native/machine-coding/blobmaker/style.css:1]: import core styles

File: native/machine-coding/blobmaker/style.css [hunk 3]
```diff
@@ -0,0 +1,103 @@
+* {
+    padding: 0;
+    margin: 0;
+    box-sizing: border-box;
+    outline: none;
+    font-family: "Roboto Mono", monospace;
+    font-weight: 400;
+    color: #010120;
+}
```
Comment (sadanandpai) [inline native/machine-coding/blobmaker/style.css:9]: * rule comes with core styles. no need to override

File: native/machine-coding/blobmaker/index.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="icon" type="image/svg+xml" href="../../logo.svg" />
+    <link rel="stylesheet" href="style.css">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@400;500&display=swap" rel="stylesheet">
```
Comment (sadanandpai) [inline native/machine-coding/blobmaker/index.html:11]: dont import additional font

File: native/machine-coding/blobmaker/app.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+let sliders = document.querySelectorAll("input[type='range']");
+    sliders.forEach(function(slider) {
+        slider.addEventListener("input", createBlob);
+    });
+
+let inputs = document.querySelectorAll("input[type='number']");
+    inputs.forEach(function(inp) {
+        inp.addEventListener("change", createBlob);
+    });
```
Comment (sadanandpai) [inline native/machine-coding/blobmaker/app.js:11]: use const if no reassignment is done

**Review by sadanandpai — CHANGES_REQUESTED** [review]

File: native/machine-coding/blobmaker/style.css
Comment (AbhineshJha) [reply native/machine-coding/blobmaker/style.css:1]: what do you mean ??

Comment (sadanandpai) [reply native/machine-coding/blobmaker/style.css:1]: please go through contributing.md and other challenges. Follow the standards

File: native/machine-coding/blobmaker/index.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="icon" type="image/svg+xml" href="../../logo.svg" />
+    <link rel="stylesheet" href="style.css">
+    <script src="../../helpers/challenges.js" type="module" defer></script>
+    <title>Blob Maker</title>
```
Comment (sadanandpai) [inline native/machine-coding/blobmaker/index.html:10]: import the navbar

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by AbhineshJha** [comment]
alright @sadanandpai

File: native/machine-coding/blobmaker/index.html [hunk 7]
```diff
@@ -0,0 +1,44 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="icon" type="image/svg+xml" href="../../logo.svg" />
+    <link rel="stylesheet" href="style.css">
+    <script src="../../helpers/challenges.js" type="module" defer></script>
```
Comment (sadanandpai) [inline native/machine-coding/blobmaker/index.html:9]: why do we need challenges.js import here

**Review by sadanandpai — APPROVED** [review]

**Pull request merged**